### DISCUSSION TITLE
A Possible Resource

### DISCUSSION THREAD

**Discussion by ctlw83** [Alovoa/alovoa#285] [General]
There is another Open Source dating platform out there.  It appears to use a VERY different design methodology and framework than Alovoa, but I am sharing the link here in case there is anything that might be cross-compatible or that the devs here could draw inspiration from, etc.

https://github.com/pH7Software/pH7-Social-Dating-CMS

Note: Me sharing this does not imply a preference on my part toward either system.  It is just meant to say "hey look, there is another thing out there you could maybe cannibalize code from"  :D

**Comment by ip6li** [comment]
I tried pH7, but it was really disappointing. High load tests failed, pH7 is good for several thousands of users, but will definitely run in problems with millions of users. It is not scalable on hyperscalers like Amazon AWS.
Alovoa has similar problems for now, but it is easier to migrate to more scalability. A SQL database scales not very well on high load platforms but Alovoa uses frameworks which can be configured to a better scalability.

In a first step I am working on some security issues (e.g. secrets storage, old dependencies etc.). Next step is to get Alovoa more stateless and testable in a distributed environment (e.g. Docker Swarm or Kubernetes). At least a review of its architecture seems to be necessary, but on a first glance its architecture looks quite healthy.

### Backends

Unfortunately a distributed environment means more dependencies:

* Secrets storage: Hashicorp Vault, possibly with HSM. My fork of Alovoa works now with Hashicorp Vault.
* Identity provider: For internal users a local Keycloak installation is a good choice. Local SQL based authentication should be disabled.
* Redis as replacement for MariaDB/MySQL at least for larger blobs.
* Reverse proxy/WAF is a must have for security and load balancing. In very large environments Cloudflare may be good choice. For small environments and tests Nginx may be sufficient.
